WPFHospitalEditor/DynamicGridControl.xaml.cs

```diff
@@ -15,13 +15,15 @@ public partial class DynamicGridControl : UserControl
         private List<Label> labels = new List<Label>();
         private List<TextBox> textBoxes = new List<TextBox>();
         private Role role;
+        private Boolean isEquipmentOrMedicine;
 
-        public DynamicGridControl(String[] contentRows, String separator, Role role)
+        public DynamicGridControl(String[] contentRows, String separator, Role role, Boolean isEquipmentOrMedicine)
```

Comment:
Do not charge the component with too much logic and parameters. It would be better if you give one parameter in DynamicGridControl that would be isReadOnly (except role and isEquipmentOrMedicine).

Reply:
fixed